File: backend/backend.Domain/src/Entities/UserCard.cs
namespace backend.Domain.src.Entities
{
    public class UserCard : BaseEntity
    {
        public string CardName { get; set; }
        public string Type { get; set; } // Mastercard or Visa
        public string CardNumber { get; set; }
        public DateOnly ExpiredDate { get; set; }
        public int CVV { get; set; }
        public Guid UserId { get; set; }
        public User User { get; set; }
    }
}

File: frontend/src/types/UserCard.ts
import { Guid } from "guid-typescript"

export interface UserCardRead
{
    id: Guid
    cardName: string
    type: string
    cardNumber: string
    expiredDate: Date
    userId: Guid
}

export interface UserCardCreate
{
    cardName: string
    type: string
    cardNumber: string
    expiredDate: Date
    cvv: number
    userId: Guid
}